core/CMakeLists.txt

```diff
@@ -918,7 +918,7 @@ copy_target_to_device(${PRELOAD_NAME} "${location_suffix}")
 # drpreinject.dll doesn't link in instr_shared.c so we can't include our inline
 # functions.
 set_target_properties(${PRELOAD_NAME} PROPERTIES
-  COMPILE_FLAGS "-DNOT_DYNAMORIO_CORE_PROPER -DRC_IS_PRELOAD -DDR_NO_FAST_IR")
+  COMPILE_DEFINITIONS "NOT_DYNAMORIO_CORE_PROPER;RC_IS_PRELOAD;DR_NO_FAST_IR")
```

Comment:
I wonder if this is expanded correctly: https://stackoverflow.com/questions/11594905/how-do-i-correctly-pass-cmake-list-semicolon-sep-of-flags-to-set-target-proper

Reply:
COMPILE_DEFINITIONS is a list, but set_target_properties won't take it as separate tokens (some other property-setting functions can) so it's passed as a list-as-string.